File: BC-W4/Army/Abilities/DemonAbility/DemonAbility.cpp
#include "DemonAbility.h"

DemonAbility::DemonAbility(Unit* unit) : Ability(unit) {};
DemonAbility::~DemonAbility() {};

void DemonAbility::castAbility() {
    throw HasNoSuchAbilityException();
};

void DemonAbility::castAbility(Unit* enemy) {
    throw HasNoSuchAbilityException();
};

File: BC-W4/ArmyJava/src/abilities/DemonAbility.java
package abilities;

import exceptions.InvalidInfectException;
import exceptions.InvalidTransformException;
import units.Unit;

public class DemonAbility extends Ability {
    public DemonAbility(Unit owner) {
        super(owner);
    }

    public void transform() throws InvalidTransformException {
        throw new InvalidTransformException();
    }

    public void infect(Unit enemy) throws InvalidInfectException {
        throw new InvalidInfectException();
    }
}
